### ISSUE TITLE
cannot add team as github owner for lalrpop crate

### ISSUE THREAD

**Issue by nikomatsakis** [rust-lang/cargo#5297]
I get strange errors, see transcript below:

```bash
>  cargo owner --add github:lalrpop:publish
    Updating registry `https://github.com/rust-lang/crates.io-index`
error: failed to invite owners to crate lalrpop: failed to get a 200 OK response, got 502
headers:
    HTTP/1.1 100 Continue



    HTTP/1.1 502 Bad Gateway

    Connection: keep-alive

    Server: nginx

    Date: Thu, 05 Apr 2018 08:47:57 GMT

    Content-Type: text/html

    Content-Length: 166

    Strict-Transport-Security: max-age=31536000

    Via: 1.1 vegur



body:
<html>
<head><title>502 Bad Gateway</title></head>
<body bgcolor="white">
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

**Comment by carols10cents** [comment]
Here's what I see in the papertrail logs searching for `crates/lalrpop/owners`:

```
Apr 05 04:37:33 crates-io app/web.2: ERROR 2018-04-05T08:37:33Z: conduit_log_requests: 127.0.0.1:51462 [2018-04-05T08:37:33Z] Put /api/v1/crates/lalrpop/owners - 70ms 500: didn't get a 200 result from github, got 401 with: {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"} didn't get a 200 result from github, got 401 with: {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"} 
Apr 05 04:37:33 crates-io app/web.2: 2018/04/05 08:37:33 [error] 100#0: *132269 upstream prematurely closed connection while reading response header from upstream, client: 10.69.109.160, server: _, request: "PUT //api/v1/crates/lalrpop/owners HTTP/1.1", upstream: "http://127.0.0.1:8888//api/v1/crates/lalrpop/owners", host: "crates.io" 
Apr 05 04:37:33 crates-io heroku/router: at=info method=PUT path="//api/v1/crates/lalrpop/owners" host=crates.io request_id=e9608c5b-92b9-4e6b-ac32-ce5f5c0ba811 fwd="209.6.50.191" dyno=web.2 connect=1ms service=90ms status=502 bytes=356 protocol=https 
Apr 05 04:47:57 crates-io heroku/router: at=info method=PUT path="//api/v1/crates/lalrpop/owners" host=crates.io request_id=ea1e4af3-3993-43fb-be9b-a4feb7e2fff0 fwd="209.6.50.191" dyno=web.2 connect=0ms service=73ms status=502 bytes=356 protocol=https 
Apr 05 04:47:57 crates-io app/web.2: ERROR 2018-04-05T08:47:57Z: conduit_log_requests: 127.0.0.1:60820 [2018-04-05T08:47:57Z] Put /api/v1/crates/lalrpop/owners - 52ms 500: didn't get a 200 result from github, got 401 with: {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"} didn't get a 200 result from github, got 401 with: {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"} 
Apr 05 04:47:57 crates-io app/web.2: 2018/04/05 08:47:57 [error] 100#0: *138593 upstream prematurely closed connection while reading response header from upstream, client: 10.151.201.12, server: _, request: "PUT //api/v1/crates/lalrpop/owners HTTP/1.1", upstream: "http://127.0.0.1:8888//api/v1/crates/lalrpop/owners", host: "crates.io" 
```

tl;dr github is saying "bad credentials" and I'm not sure why. 

An unsatisfying potential fix: could you revoke crates.io's oauth permissions from your github account via https://github.com/settings/applications, log out of crates.io in your browser, log back in doing the oauth dance again, and then trying cargo owner add again?

**Comment by nikomatsakis** [comment]
@carols10cents I don't see `crates.io` in that list...

**Comment by carols10cents** [comment]
@nikomatsakis hmmm have you ever signed into crates.io in the browser??? could you try :)

**Comment by nikomatsakis** [comment]
I think I never have. I'm trying that now as we type =)

**Comment by brendanzab** [comment]
Just a heads up, I also ran into this, and have been confused for the last few weeks until I found this issue today! I tried:

```
$ cargo owner --add github:yeslogic:developers-fathom
    Updating crates.io index
error: failed to invite owners to crate fathom: api errors (status 200 OK): could not find the github team yeslogic/developers-fathom
```

Going to https://github.com/settings/applications, I noticed that crates.io was not in the list. I went to https://crates.io/login and logged in, authorising crates.io. Retrying the original command I got this:

```
$ cargo owner --add github:yeslogic:developers-fathom
    Updating crates.io index
       Owner team github:yeslogic:developers-fathom has been added as an owner of crate fathom
```

:tada:

It would be really nice if the original error was clearer! I originally thought it might have been something to do with syncing the github team with crates.io, but that was a red herring! 😅
